worldguard-bukkit/src/main/java/com/sk89q/worldguard/bukkit/listener/BlacklistListener.java

```diff
@@ -96,7 +96,7 @@ public void onBreakBlock(final BreakBlockEvent event) {
         event.filter(target -> {
             if (!wcfg.getBlacklist().check(
                     new BlockBreakBlacklistEvent(localPlayer, BukkitAdapter.asBlockVector(target),
-                            createTarget(target.getBlock(), event.getEffectiveMaterial())), false, false)) {
+                            createTarget(target.getBlock())), false, false)) {
```

Comment:
Instead of using the effectiveMaterial I changed this to the actual block for on-break, on-place and on-interact. This has some implications:

* Now the actual block is checked. If someone is destroying the world with TNT the destroyed block type is checked and not the TNT material.
* on-use can be used to check to deny usage of stuff. e.g. on-use[WIND_CHARGE] will block the usage of wind charges, on-interact[OAK_DOOR] will block the interaction of the wind charge with the door.